### ISSUE TITLE
Incorrect behavior of "azure-iot-device-mqtt.clientFromConnectionString()" with a wrong connection string

### ISSUE THREAD

**Issue by Voxiferus** [Azure/azure-iot-sdk-node#10]
# Context
- **OS and version used:** Windows 7
- **Node.js version:** 6.3.0
- **npm version:** 3.10.3
- **list of installed packages:**
    "azure-iot-device": "^1.1.2",
    "azure-iot-device-mqtt": "^1.1.2",

# Description of the issue:
### Incorrect behavior of "azure-iot-device-mqtt.clientFromConnectionString()" with a wrong connection string.

With some wrong but parsable connection string   .open() method's callback fires twice.

If the value of "HostName" is correct but "SharedAccessKey" is not, it gets "Connection refused: Not authorized" error for the first call, and NULL for the second one, what looks like a successful connection established.

For sure, further manipulations with client object, in this case, are not possible, e.g. client.sendEvent() callback will never be fired.

# Code sample exhibiting the issue: 
```
var clientFromConnectionString = require('azure-iot-device-mqtt').clientFromConnectionString;

var client = clientFromConnectionString('HostName=correct-host-name.azure-devices.net;DeviceId=something;SharedAccessKey=some-wrong-key');

client.open((err) => {
    if (err) {
        console.error(err.message);
    } else {
        console.log('Client connected');
    }
});
```

# Console log of the issue:  
```
Connection refused: Not authorized
Client connected
```

**Comment by anhashia** [comment]
Investigating !

**Comment by anhashia** [comment]
Hi @Voxiferus 
I was able to locally reproduce the issue. Issue seems specific to MQTT protocol as there are 2 callback happening for MQTT for incorrect connection string. This issue needs further investigation.

For now, you can use following **workaround** for the issue. Basically. you also check for the result code specifically for 2nd callback to avoid this issue.

**Workaround**

``````
var connectCallback = function (err,result) {
  if (err || !result) {
    if (err)
     console.error('Could not connect: ' + err.message);
    else
     console.error('Could not connect: No result ' );
  } else {
    console.log('Client connected');
``````

**Comment by pierreca** [comment]
Got down to the bottom of the issue which was the mqtt library firing multiple events, leading us to call the callback multiple times. Fix just got checked in (and referenced above). we will release a new set of packages before the end of the month.

**Comment by pierreca** [comment]
The fix has been released to npm with packages version 1.1.4:
- https://www.npmjs.com/package/azure-iot-device-mqtt
- https://www.npmjs.com/package/azure-iot-device
